query/src/main/scala/filodb/query/exec/rangefn/AggrOverTimeFunctions.scala

```diff
@@ -85,12 +121,36 @@ class AvgOverTimeFunction extends RangeFunction {
   }
 }
 
-class StdDevOverTimeFunction extends RangeFunction {
+abstract class AvgOverTimeChunkedFunction(var sum: Double = 0d, var count: Int = 0) extends ChunkedRangeFunction {
+  override final def reset(): Unit = { sum = 0d; count = 0 }
+  final def apply(endTimestamp: Long, sampleToEmit: TransientRow): Unit = {
+    sampleToEmit.setValues(endTimestamp, if (count > 0) sum/count else 0d)
+  }
+}
+
+class AvgOverTimeChunkedFunctionD extends AvgOverTimeChunkedFunction() with ChunkedDoubleRangeFunction {
+  final def addTimeDoubleChunks(doubleVect: BinaryVector.BinaryVectorPtr,
+                                doubleReader: bv.DoubleVectorDataReader,
+                                startRowNum: Int,
+                                endRowNum: Int): Unit = {
+    sum += doubleReader.sum(doubleVect, startRowNum, endRowNum)
+    count += (endRowNum - startRowNum + 1)
```

Comment:
NaN needs to be handled for both sum and count

Reply:
Yes, will do.